catroid/src/main/java/org/catrobat/catroid/content/actions/HideTextAction.java

```diff
@@ -37,20 +37,22 @@
 public class HideTextAction extends TemporalAction {
 
 	private UserVariable variableToHide;
+
+	private Scene scene;
+	private Sprite sprite;
 	private UserBrick userBrick;
 
 	@Override
 	protected void begin() {
-		DataContainer dataContainer = ProjectManager.getInstance().getCurrentScene().getDataContainer();
-		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
+		DataContainer dataContainer = scene.getDataContainer();
```

Comment:
I agree with @loibi93, `getSceneOfSprite` seems to be unnecessary.
Since it should be merged to the NOLP branch asap, I'm also fine to use `getSceneToPlay()` here.

However, DI as you already use it [here](https://github.com/Catrobat/Catroid/pull/2021/files#diff-a353972b5dc62a8ccef1151508102651R922) is a cleaner solution. You should also be able use `getSceneToPlay` there. The best solution would be to pass the parameter Scene all from the start but this'll be a bigger refactoring. Also I can't guarantee if it'll work as expected. :wink: